validator_client/src/http_api/mod.rs

```diff
@@ -1028,6 +1034,116 @@ pub fn serve<T: 'static + SlotClock + Clone, E: EthSpec>(
+                            "An error occurred while attempting to get graffiti".into(),
+                        ))
+                    }
+                })
+            },
+        );
+
+    // POST /eth/v1/validator/{pubkey}/graffiti
+    let post_graffiti = eth_v1
+        .and(warp::path("validator"))
+        .and(warp::path::param::<PublicKey>())
+        .and(warp::path("graffiti"))
+        .and(warp::body::json())
+        .and(warp::path::end())
+        .and(validator_store_filter.clone())
+        .and(graffiti_file_filter.clone())
+        .and(signer.clone())
+        .and(task_executor_filter.clone())
+        .and_then(
+            |pubkey: PublicKey,
+             query: SetGraffitiRequest,
+             validator_store: Arc<ValidatorStore<T, E>>,
+             graffiti_file: Option<GraffitiFile>,
+             signer,
+             task_executor: TaskExecutor| {
+                blocking_signed_json_task(signer, move || {
+                    if graffiti_file.is_some() {
+                        return Err(warp_utils::reject::invalid_auth(
+                            "Unable to update graffiti as the \"--graffiti-file\" flag is set"
+                                .to_string(),
+                        ));
+                    }
+                    if let Some(handle) = task_executor.handle() {
+                        let Some(graffiti) = query.graffiti else {
+                            return Err(warp_utils::reject::custom_bad_request(
+                                "No graffiti provided".into(),
+                            ));
+                        };
+                        handle.block_on(set_graffiti(pubkey.clone(), graffiti, validator_store))?;
+                        Ok(())
+                    } else {
+                        Err(warp_utils::reject::custom_server_error(
+                            "An error occurred while attempting to set graffiti".into(),
+                        ))
+                    }
+                })
+            },
+        )
+        .map(|reply| warp::reply::with_status(reply, warp::http::StatusCode::ACCEPTED));
+
+    // DELETE /eth/v1/validator/{pubkey}/graffiti
+    let delete_graffiti = eth_v1
+        .and(warp::path("validator"))
+        .and(warp::path::param::<PublicKey>())
+        .and(warp::path("graffiti"))
+        .and(warp::path::end())
+        .and(validator_store_filter.clone())
+        .and(graffiti_file_filter.clone())
+        .and(signer.clone())
+        .and(task_executor_filter.clone())
+        .and_then(
+            |pubkey: PublicKey,
+             validator_store: Arc<ValidatorStore<T, E>>,
+             graffiti_file: Option<GraffitiFile>,
+             signer,
+             task_executor: TaskExecutor| {
+                blocking_signed_json_task(signer, move || {
+                    if graffiti_file.is_some() {
+                        return Err(warp_utils::reject::invalid_auth(
```

Comment:
It's a little odd that we use 403 for this but I think this is correct according to spec.. I think adding an extra type for this might be unnecessary other than slight improve in readability though.

Reply:
agreed that 403 is a bit odd here. what do you mean by an extra type?

Reply:
do you mean maybe adding a new `forbidden` type, or something along those lines?

Reply:
yeah, that's what I was thinking about, but `forbidden` kind of mean the same thing - i think it's probably ok to leave it as it is, I don't think it adds much value covering the fact that we're returning a 403.